packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/inspections/impl/FieldCheckLinterInspectionTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.mongodb.jbplugin.inspections.impl
+
+import com.intellij.database.dataSource.DatabaseConnectionManager
+import com.intellij.database.dataSource.DatabaseConnectionPoint
+import com.intellij.database.dataSource.localDataSource
+import com.intellij.database.psi.DbDataSource
+import com.intellij.database.psi.DbPsiFacade
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+import com.mongodb.jbplugin.accessadapter.datagrip.DataGripBasedReadModelProvider
+import com.mongodb.jbplugin.accessadapter.slice.GetCollectionSchema
+import com.mongodb.jbplugin.editor.MongoDbVirtualFileDataSourceProvider
+import com.mongodb.jbplugin.fixtures.*
+import com.mongodb.jbplugin.fixtures.mockDataSource
+import com.mongodb.jbplugin.inspections.bridge.FieldExistenceCheckInspectionBridge
+import com.mongodb.jbplugin.mql.BsonObject
+import com.mongodb.jbplugin.mql.CollectionSchema
+import com.mongodb.jbplugin.mql.Namespace
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+
+@CodeInsightTest
+@Suppress("TOO_LONG_FUNCTION", "LONG_LINE")
+class FieldCheckLinterInspectionTest {
+    @ParsingTest(
+        fileName = "Repository.java",
+        value = """
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import static com.mongodb.client.model.Filters.*;
+
+public class Repository {
+    private final MongoClient client;
+
+    public Repository(MongoClient client) {
+        this.client = client;
+    }
+
+    public FindIterable<Document> exampleFind() {
+        return <warning descr="No connection available to run this check.">client.getDatabase("myDatabase")
+                .getCollection("myCollection")
+                .find()</warning>;
+    }
+}
+        """,
+    )
+    fun `shows an intention when there is no connection attached to the current editor`(
```

Comment:
```suggestion
    fun `shows an inspection when there is no connection attached to the current editor`(
```
I guess? :D The same below if it applies